tests/test_main.py

```diff
@@ -0,0 +1,265 @@
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_set_priority_not_priority_more_than_max_priority(self, MockPullReqState):
+        state = MockPullReqState()
+        set_priority(state, True, 'p=5', {'max_priority': 3})
+        state.add_comment.assert_called_once_with(':stop_sign: Priority higher than 3 is ignored.')
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_approver_me(self, MockPullReqState):
+        state = MockPullReqState()
+        self.assertFalse(review_approved(['r=me', 'approved'], 'r=me', state, True, 'user', 'user', 0, '', []))
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_wip(self, MockPullReqState):
+        state = MockPullReqState()
+        state.title = 'WIP work in progress'
+        self.assertFalse(review_approved(['r+', 'approved'], 'r+', state, True, 'user', 'user', 0, '', []))
+        state.add_comment.assert_called_once_with(':clipboard: Looks like this PR is still in progress, ignoring approval')
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_todo(self, MockPullReqState):
+        state = MockPullReqState()
+        state.title = 'TODO work in progress'
+        self.assertFalse(review_approved(['r+', 'approved'], 'r+', state, True, 'user', 'user', 0, '', []))
+        state.add_comment.assert_called_once_with(':clipboard: Looks like this PR is still in progress, ignoring approval')
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_equal_usernames(self, MockPullReqState):
+        state = MockPullReqState()
+        state.head_sha = 'abcd123'
+        state.title = "My pull request"
+        self.assertTrue(review_approved(['r+', 'approved'], 'r+', state, True, 'user', 'user', 0, 'abcd123', []))
+        self.assertEqual(state.approved_by, 'user')
+        self.assertFalse(state.try_)
+        state.set_status.assert_called_once_with('')
+        state.save.assert_called_once()
+
+    @patch('homu.main.PullReqState')
+    def test_review_approved_different_usernames_sha_equals_head_sha(self, MockPullReqState):
+        state = MockPullReqState()
+        state.head_sha = 'abcd123'
+        state.title = "My pull request"
+        state.repo_label = 'label'
+        state.status = 'pending'
+        states = {}
+        states[state.repo_label] = {'label': state}
+        self.assertTrue(review_approved(['r+', 'approved'], 'r+', state, True, 'user1', 'user2', 0, 'abcd123', states))
+        self.assertEqual(state.approved_by, 'user1')
+        self.assertFalse(state.try_)
+        state.set_status.assert_called_once_with('')
+        state.save.assert_called_once()
+        state.add_comment.assert_called_once_with(":bulb: This pull request was already approved, no need to approve it again.\n\n- This pull request is currently being tested. If there's no response from the continuous integration service, you may use `retry` to trigger a build again.")
```

Comment:
It might be nice to have these strings in a common variable that can be used in the implementation and the test.